src/render/tutorials/caching-http-requests/2_0.html.md

```diff
@@ -0,0 +1,328 @@
+
+```yaml
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: API gateway caching example
+  description: API gateway caching example
+
+servers:
+  - url: https://{domain}:{port}
+    description: The local API server
+    variables:
+      domain:
+        default: localhost
+        description: api domain
+      port:
+        enum:
+          - '8092'
+        default: '8092'
+
+paths:
+  /healthcheck:
+    get:
+      operationId: healthcheck-operation
+      responses:
+        default:
+          description: example vert.x healthcheck
+
+  /product/id:
+    get:
+      operationId: product-api-caching-proxy-operation
+      responses:
+        default:
+          description: External API (Product API) caching proxy.
+```
+The API is exposed by the `product/id` path and GET HTTP method. It delegates its processing logic to the `product-api-caching-proxy-operation` operation.  
+
+We have all Knot.x specific files created, now we are going to focus on project setup and deployment.  
+
+## Environment configuration
+The first thing we need to do is to specify configuration files. Let's start in the project level directory and simply create `api-cache.yml`
+This file is responsible for proper Docker configuration, in our case we are using Docker Swarm, therefore we have to configure
+two images. Swarm is needed here because we need to expose external API (it's mocked by Wiremock in this case) and the API available for user.
+Both are deployed in separate containers, therefore orchestrator is needed.
+Let's specify the following configuration:
+``` yml
+version: '3.7'
+
+networks:
+  knotnet:
+
+services:
+
+  webapi:
+    image: rodolpheche/wiremock
+    volumes:
+      - "./common-services/webapi:/home/wiremock"
+    ports:
+      - "3000:8080"
+    networks:
+      - knotnet
+
+  knotx:
+    image: knotx-example/api-cache:latest
+    command: ["knotx", "run-knotx"]
+    ports:
+      - "8092:8092"
+      - "18092:18092"
+    networks:
+      - knotnet
+```
+
+First and foremost it specifies Docker virtual network in which our services will be deployed. Then we have some service configs.
+The first one is `webapi`
+- `image` - docker image is provided here, 
+- `volumes` - directories to which Docker will have access, in our case it's directory where all Wiremock configs are stored,
+- `ports` - port accessible from `localhost` is specified on the left side, and a port available in the virtual environment is specified on the right 
+- `networks` - specifies in which docker network the service will be deployed.
+
+Then we have `knotx` service specification. It's very similar to the previous one, so only one property is worth explaining. 
+- `command` - this property is used to specify Knot.x starting commands.
+
+Next step in our configuration is to set `rootProject.name` in `settings.gradle.kts` (also at the project level directory),
```

Comment:
I am not sure if this rename is required for this tutorial. The "api-cache" name is not connected with docker image name.

Reply:
it will of course work without it, but I think that we can provide proper name for it as it is